### ISSUE TITLE
[MockFileSystem] Directory discover could return the directory self

### ISSUE THREAD

**Issue by zvirja** [tathamoddie/System.IO.Abstractions#169]
Consider the following code:

``` csharp
      var mfs = new MockFileSystem();
      mfs.Directory.CreateDirectory("dir");
      var directoriesRelative = mfs.Directory.GetDirectories("dir");
      var directoriesAbs = mfs.Directory.GetDirectories(mfs.Path.GetFullPath("dir"));
```

You will see that `directoriesRelative` contains the `dir` folder itself, while `directoriesAbs` is empty as expected.

This is a critical bug because it forces to rewrite production code to cover it later by tests.

Likely, in order to fix the issue the following two lines should be reordered: [MockDirectory.cs](https://github.com/tathamoddie/System.IO.Abstractions/blob/master/TestingHelpers/MockDirectory.cs#L180).

**Comment by MCCoolMoDAue** [comment]
Hi Zvirja, I am looking into your issue.
Can you elaborate which lines you mean?
only one line is highighted

**Comment by Zvirja** [comment]
Good catch :) I believe that [these two lines](https://github.com/tathamoddie/System.IO.Abstractions/blob/master/TestingHelpers/MockDirectory.cs#L180-L181) should be reordered among themselves.

Will be really thankful if this bug is fixed.

**Comment by tathamoddie** [comment]
Fixed in #258, shipped to NuGet as 2.1.0.174
